src/twenty-three-video/twenty-three-video.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { DisplayApiService } from "../display-api/display-api.service";
+import { HttpService } from "@nestjs/axios";
+import { Slide } from "../display-api/types";
+import { lastValueFrom } from "rxjs";
+import { SlideSlideInput } from "../display-api-client";
+
+interface TwentyThreeVideoListParams {
+  format: string;
+  raw: boolean;
+  tags?: string;
+  album_id?: string;
+}
+
+@Injectable()
+export class TwentyThreeVideoService {
+  constructor(
+    private readonly logger: Logger,
+    private readonly httpService: HttpService,
+    private readonly displayApi: DisplayApiService
+  ) {}
+
+  // TODO, move this out into configuration.
+  private readonly baseUrl = "https://video.kk.dk/api/photo/list";
+  private readonly slideType = "TwentyThreeVideo";
+
+  async syncAllSlides(): Promise<void> {
+    this.logger.debug("TwentyThreeVideoService::syncAllSlides");
+    const slides = await this.displayApi.fetchSlides(this.slideType);
+    for (const slide of slides) {
+      await this.syncSlide(slide);
+    }
+  }
+
+  async syncSlide(slide: Slide): Promise<void> {
+    const args: TwentyThreeVideoListParams = {
+      format: "json",
+      raw: true,
+    };
+    if (slide.content.tags) {
+      args.tags = slide.content.tags;
+    }
+    if (slide.content.album_id) {
+      args.album_id = slide.content.album_id;
+    }
+    const jsonData = await this.fetchTwentyThreeVideoList(args);
+    if (Array.isArray(jsonData)) {
+      slide.content.jsonData = JSON.stringify(jsonData);
+      const id = slide["@id"].split("/").slice(-1)[0];
+      await this.displayApi.updateSlide(id, slide as unknown as SlideSlideInput);
+    }
+  }
+
+  async fetchTwentyThreeVideoList(params: TwentyThreeVideoListParams): Promise<Array<any> | void> {
+    try {
+      const data = await lastValueFrom(this.httpService.get(this.baseUrl, { params })).then(
+        (response) => response.data
+      );
+      if (data.status !== "ok") {
+        this.logger.error("❌ ~ error fetching kk video list did not get ok status back");
```

Comment:
It's unnecessary to prefix the errors with `❌ ~ error ` - if we want something like that we should add it to a console-specific log formatter. This way we'll have emojis in all sorts of lowlevel logs.

Reply:
I realise that I should have mentioned this earlier and we now have this pattern in other parts of the code. Maybe we should do the cleanup in a seperate PR?

Reply:
I think I saw this pattern in some code you pointed :)

Reply:
Probably some of the poc code then 🤞 :)